Backend/app/dashboard/provider-dashboard/src/API/Exotel.hs

```diff
@@ -0,0 +1,83 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module API.Exotel
+  ( API,
+    handler,
+  )
+where
+
+import qualified "dashboard-helper-api" Dashboard.Common.Exotel as Common
+import qualified Domain.Types.ServerName as DSN
+import qualified Domain.Types.Transaction as DT
+import "lib-dashboard" Environment
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Types.APISuccess
+import Kernel.Utils.Common (MonadFlow, decodeFromText, fork, withFlowHandlerAPI)
+import qualified ProviderPlatformClient.DynamicOfferDriver as Client
+import qualified ProviderPlatformClient.StaticOfferDriver as Client
+import qualified RiderPlatformClient.RiderApp as Client
+import Servant
+import qualified SharedLogic.Transaction as T
+import qualified Storage.Queries.Transaction as QT
+
+type API =
+  "exotel"
+    :> Common.ExotelHeartbeatAPI
+
+handler :: FlowServer API
+handler =
+  exotelHeartbeat
+
+buildTransaction ::
+  ( MonadFlow m,
+    Common.HideSecrets request
+  ) =>
+  Common.ExotelEndpoint ->
+  DSN.ServerName ->
+  Maybe request ->
+  m DT.Transaction
+buildTransaction endpoint serverName =
+  T.buildTransaction (DT.ExotelAPI endpoint) (Just serverName) Nothing Nothing Nothing
+
+-- store request and call bap/bpp only when status is not OK or previous status was not OK
+exotelHeartbeat :: Common.ExotelHeartbeatReq -> FlowHandler APISuccess
+exotelHeartbeat req = withFlowHandlerAPI $ do
+  forM_ [DSN.APP_BACKEND, DSN.BECKN_TRANSPORT, DSN.DRIVER_OFFER_BPP] $ \serverName -> do
+    fork ("exotelHeartbeat:" <> show serverName) $ do
+      needToCallApp <-
+        if req.statusType /= Common.OK
+          then pure True
```

Comment:
Here also we can have check if last status is OK then only return True. So, ideally we should notify BAP/BPP if status is changing from OK to not OK and vice versa.

Reply:
But what if we have status not OK for some other phone number, like:

request1:
```
{  
   "timestamp": "2018-08-22T15:19:23Z>",
   "status_type":"WARNING",
   "incoming_affected":[ 
      "07930061010"
   ],
   "outgoing_affected": null,
   "data":{  
      "07930061010":{  
         "phone_number": "+917930061010",
      }
   }
}
```

request2:

```
{  
   "timestamp": "2018-08-22T15:19:23Z>",
   "status_type":"WARNING",
   "incoming_affected":[ 
      "someOtherNumber"
   ],
   "outgoing_affected": null,
   "data":{  
      "07930061010":{  
         "phone_number": "someOtherNumber",
      }
   }
}
```

Maybe we should store all requests which are not OK?

Reply:
We can notify bap/bpp every time when something changed. Like, if other phone number was affected, or status changed from not OK to OK. What do you think @hkmangla ?

Reply:
> We can notify bap/bpp every time when something changed. Like, if other phone number was affected, or status changed from not OK to OK.

Implemented this one and rebased. Please check and merge.